explore/src/main/scala/explore/itc/ItcPanelTitle.scala

```diff
@@ -60,53 +69,77 @@ object ItcPanelTitle:
-            .setState(Pot.error(new RuntimeException("Not enough information to call ITC")))
+            .setState(
+              props.itcTargets.foldMap(_.toList.map(t => t -> MissingInfo)).toMap
+            )
             .to[IO],
           IO.unit
         )
       }
       .render { (props, results) =>
-        def newSelected(p: ItcTarget): Option[ItcTarget] =
-          props.targets.find(_ === p)
+        def newSelected(p: Int): Option[ItcTarget] =
+          props.targets.lift(p)
+
+        val selectedResult: Pot[ItcChartResult] =
+          Pot
+            .fromOption(props.selectedTarget.get)
+            .flatMap(t => results.value.getOrElse(t, Pot.pending[ItcChartResult]))
+
+        val selected       = props.selectedTarget.get.map(_.name.value)
+        val selectedTarget = props.selectedTarget.get
 
-        val selectedResult: Option[ItcChartResult] =
-          props.selectedTarget.get.flatMap(t => results.value.toOption.flatMap(_.get(t)))
+        val itcTargets          = props.itcTargets.foldMap(_.toList)
+        val idx                 = itcTargets.indexWhere(selectedTarget.contains)
+        val itcTargetsWithIndex = itcTargets.zipWithIndex
 
-        val selected = props.selectedTarget.get.map(_.name.value)
+        val ccds                                 = selectedResult.map(_._2)
+        def singleSN: ItcChartResult => VdomNode =
+          (r: ItcChartResult) => <.span(formatCcds(r.ccds.some, _.maxSingleSNRatio.toString))
+        def totalSN: ItcChartResult => VdomNode  =
+          (r: ItcChartResult) => <.span(formatCcds(r.ccds.some, _.maxTotalSNRatio.toString))
+        val existTargets                         = props.targets.nonEmpty
 
-        val itcTargets = props.itcTargets.foldMap(_.toList)
-        val ccds       = selectedResult.map(_._2)
-        val singleSN   = formatCcds(ccds, _.maxSingleSNRatio.toString)
-        val totalSN    = formatCcds(ccds, _.maxTotalSNRatio.toString)
+        def snSection(title: String, fn: ItcChartResult => VdomNode) =
+          React.Fragment(
+            <.label(title),
+            if (existTargets) {
+              potRender[ItcChartResult](
+                singleSN,
+                Icons.Spinner.spin(true),
+                e => <.span(MissingInfoIcon).withTooltip(e.getMessage)
+              )(
+                selectedResult
+              )
+            } else {
+              <.span(MissingInfoIcon).withTooltip(MissingInfoMsg)
+            }
+          )
 
         <.div(
           ExploreStyles.ItcTileTitle,
           <.label(s"Target:"),
           Dropdown(
             clazz = ExploreStyles.ItcTileTargetSelector,
-            value = props.selectedTarget.get.orUndefined,
+            value = idx,
             onChange = {
-              case t @ ItcTarget(_, _, _) => props.selectedTarget.set(newSelected(t))
-              case _                      => Callback.empty
+              case t: Int => props.selectedTarget.set(newSelected(t))
```

Comment:
Why was it necessary to switch to value to an index? Just to know if this is something to consider in the future.

Reply:
Right, this was tricky.
The object set in value is compared (I think) internally as js objects which make it not match as you'd expect a scala object .
We may want to have a `Droopdown[A]` eventually

Reply:
I have already changed it to `Dropdown[A]` in my recent work to make it safer. There will be a `Dropdown[A]` and a `DropdownOptional[A]`

Reply:
Oh great, would it it take an `Eq[A]`?

Reply:
When I upgrade explore to the new lucuma-react, I'll see if this works correctly.

Reply:
> Oh great, would it it take an `Eq[A]`?

Hmm. It doesn't at the moment. So, the problem you are having is that the correct `SelectItem` isn't selected for the target?